subprojects/dependency-management/src/main/java/org/gradle/internal/component/external/model/ivy/RealisedIvyModuleResolveMetadata.java

```diff
@@ -90,31 +92,38 @@ private static void addVariantsFromRules(DefaultIvyModuleResolveMetadata compone
         }
 
         for (AdditionalVariant additionalVariant : additionalVariants) {
-            String name = additionalVariant.getName();
-            String baseName = additionalVariant.getBase();
-            ImmutableAttributes attributes;
-            ImmutableCapabilities capabilities;
-            List<ModuleDependencyMetadata> dependencies;
-            ImmutableList<? extends ModuleComponentArtifactMetadata> artifacts;
-
-            ModuleConfigurationMetadata baseConf = (ModuleConfigurationMetadata) declaredConfigurations.get(baseName);
-            if (baseConf == null) {
-                attributes = componentMetadata.getAttributes();
-                capabilities = ImmutableCapabilities.EMPTY;
-                dependencies = ImmutableList.of();
-                artifacts = ImmutableList.of();
-            } else {
-                attributes = baseConf.getAttributes();
-                capabilities = (ImmutableCapabilities) baseConf.getCapabilities();
-                dependencies = Cast.uncheckedCast(baseConf.getDependencies());
-                artifacts = Cast.uncheckedCast(baseConf.getArtifacts());
-            }
+                String name = additionalVariant.getName();
+                String baseName = additionalVariant.getBase();
+                ImmutableAttributes attributes;
+                ImmutableCapabilities capabilities;
+                List<ModuleDependencyMetadata> dependencies;
+                ImmutableList<? extends ModuleComponentArtifactMetadata> artifacts;
+                ImmutableList<ExcludeMetadata> excludes;
+
+                ModuleConfigurationMetadata baseConf = (ModuleConfigurationMetadata) declaredConfigurations.get(baseName);
+                if (baseConf == null) {
+                    attributes = componentMetadata.getAttributes();
+                    capabilities = ImmutableCapabilities.EMPTY;
+                    dependencies = ImmutableList.of();
+                    artifacts = ImmutableList.of();
+                    excludes = ImmutableList.of();
+                } else {
+                    attributes = baseConf.getAttributes();
+                    capabilities = (ImmutableCapabilities) baseConf.getCapabilities();
+                    dependencies = Cast.uncheckedCast(baseConf.getDependencies());
+                    artifacts = Cast.uncheckedCast(baseConf.getArtifacts());
+
+                    excludes = Cast.uncheckedCast(ImmutableList.of(baseConf.getExcludes().stream().map((exclude) ->
```

Comment:
@ljacomet , I tried to take the excludes and add them to the new configurations but also changing the exclude rule to point to this configuration (compile -> apiElements) for example.

It seems that this code is only hit when `org.gradle.integtest.force.realize.metadata` is present in tests.

I might be missing something here.

Will try to look into other places where the change might be needed? maybe it isn't here?